pandas/core/categorical.py

```diff
@@ -1660,16 +1660,26 @@ def fillna(self, value=None, method=None, limit=None):
 
         else:
 
-            if not isna(value) and value not in self.categories:
-                raise ValueError("fill value must be in categories")
+            if isinstance(value, ABCSeries):
```

Comment:
this could be a dict as well (which you can simply convert to a series)

Reply:
A dict has already been converted to a Series in the ``fillna`` function in ``pandas/core/generic.py``. 

So, as far as I can tell, by the time we reach the ``fillna`` function in ``pandas/core/categorical.py`` the ``value`` is either:
1.  A Series (if the user passed a Series or dict)
2. A scalar (if the user passed a scalar)

Reply:
ok, can you indicate that in a comment

Reply:
can you add a comment here? otherwise lgtm.